.github/update-sync-param-files.py

```diff
@@ -0,0 +1,39 @@
+import argparse
+from pathlib import Path
+
+import yaml
+
+
+def src_name_to_dst_name(src: Path) -> Path:
+    module_launch_pkg_name = src.parents[len(src.parents) - 3].stem  # e.g. "tier4_control_launch"
+    file_path_under_config = src.relative_to(Path("launch") / module_launch_pkg_name / "config")
+    dst = Path("autoware_launch/config") / module_launch_pkg_name / file_path_under_config
+
+    return dst
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("input_universe_dir", type=str, help="path to the target autoware.universe")
+    parser.add_argument("output_file_path", type=str, help="output path of sync-param-files.yaml")
+    args = parser.parse_args()
+
+    # iterate over the param files to create sync-param-files.yaml
+    sync_files_config = []
+    autoware_launch_dir = Path(args.input_universe_dir) / "launch"
+    for config_path in autoware_launch_dir.glob("**/*.param.yaml"):
+        src = config_path.relative_to(args.input_universe_dir)
+        dst = src_name_to_dst_name(src)
+
+        sync_files_config.append({"source": str(src), "dest": str(dst)})
+    sync_param_file = [
+        {"repository": "autowarefoundation/autoware.universe", "files": sync_files_config}
+    ]
+
+    # save sync-param-files.yaml
+    with open(args.output_file_path, "w") as f:
+        yaml.dump(sync_param_file, f, sort_keys=False)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
How about like this?

```python
import argparse
import dataclasses
from pathlib import Path
from typing import List
from typing import Optional

import git

"""
This module updates `sync-param-files.yaml` based on the launch parameter files in `autoware.universe`.
"""

REPO_NAME = "autowarefoundation/autoware.universe"
REPO_URL = f"https://github.com/{REPO_NAME}.git"
CLONE_PATH = Path("/tmp/autoware.universe")


@dataclasses.dataclass
class FileSyncConfig:
    source: str
    dest: str
    replace: Optional[bool] = None
    delete_orphaned: Optional[bool] = None
    pre_commands: Optional[str] = None
    post_commands: Optional[str] = None


def create_tier4_launch_sync_configs(tier4_launch_package_path: Path) -> List[FileSyncConfig]:
    launch_package_name = tier4_launch_package_path.name
    launch_config_path = tier4_launch_package_path / "config"

    sync_configs = []
    for param_file_path in tier4_launch_package_path.glob("config/**/*.param.yaml"):
        relative_param_file_path = param_file_path.relative_to(launch_config_path)

        source = param_file_path.relative_to(CLONE_PATH)
        dest = Path("autoware_launch/config") / launch_package_name / relative_param_file_path

        sync_configs.append(FileSyncConfig(str(source), str(dest)))

    return sync_configs


def dump_sync_config(section_name: str, sync_configs: List[FileSyncConfig]) -> List[str]:
    indent = 4 * " "
    lines = [f"{indent}# {section_name}\n"]
    for sync_config in sync_configs:
        lines.append(f"{indent}- source: {sync_config.source}\n")
        lines.append(f"{indent}  dest: {sync_config.dest}\n")
    lines.append("\n")
    return lines


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("sync_param_files_path", type=Path, help="path to sync-param-files.yaml")
    args = parser.parse_args()

    # Clone Autoware
    if not CLONE_PATH.exists():
        git.Repo.clone_from(REPO_URL, CLONE_PATH)

    # Create sync config for tier4_*_launch
    tier4_launch_package_paths = CLONE_PATH.glob("launch/tier4_*_launch")
    tier4_launch_sync_configs_map = {
        p.name: create_tier4_launch_sync_configs(p) for p in tier4_launch_package_paths
    }

    # Create sync-param-files.yaml
    with open(args.sync_param_files_path, "w") as f:
        f.write(f"- repository: {REPO_NAME}\n")
        f.write(f"  files:\n")
        for section_name, sync_config in tier4_launch_sync_configs_map.items():
            f.writelines(dump_sync_config(section_name, sync_config))


if __name__ == "__main__":
    main()
```

Reply:
The generated file is as follows:

```yaml
- repository: autowarefoundation/autoware.universe
  files:
    # tier4_localization_launch
    - source: launch/tier4_localization_launch/config/localization_error_monitor.param.yaml
      dest: autoware_launch/config/tier4_localization_launch/localization_error_monitor.param.yaml
    - source: launch/tier4_localization_launch/config/crop_box_filter_measurement_range.param.yaml
      dest: autoware_launch/config/tier4_localization_launch/crop_box_filter_measurement_range.param.yaml
    - source: launch/tier4_localization_launch/config/random_downsample_filter.param.yaml
      dest: autoware_launch/config/tier4_localization_launch/random_downsample_filter.param.yaml
    - source: launch/tier4_localization_launch/config/ndt_scan_matcher.param.yaml
      dest: autoware_launch/config/tier4_localization_launch/ndt_scan_matcher.param.yaml
    - source: launch/tier4_localization_launch/config/voxel_grid_filter.param.yaml
      dest: autoware_launch/config/tier4_localization_launch/voxel_grid_filter.param.yaml

    # tier4_sensing_launch

    # tier4_vehicle_launch

    # tier4_map_launch

    # tier4_autoware_api_launch

    # tier4_control_launch
    - source: launch/tier4_control_launch/config/common/nearest_search.param.yaml
      dest: autoware_launch/config/tier4_control_launch/common/nearest_search.param.yaml
    - source: launch/tier4_control_launch/config/operation_mode_transition_manager/operation_mode_transition_manager.param.yaml
      dest: autoware_launch/config/tier4_control_launch/operation_mode_transition_manager/operation_mode_transition_manager.param.yaml
    - source: launch/tier4_control_launch/config/vehicle_cmd_gate/vehicle_cmd_gate.param.yaml
      dest: autoware_launch/config/tier4_control_launch/vehicle_cmd_gate/vehicle_cmd_gate.param.yaml
    - source: launch/tier4_control_launch/config/trajectory_follower/latlon_muxer.param.yaml
      dest: autoware_launch/config/tier4_control_launch/trajectory_follower/latlon_muxer.param.yaml
    - source: launch/tier4_control_launch/config/trajectory_follower/longitudinal_controller.param.yaml
      dest: autoware_launch/config/tier4_control_launch/trajectory_follower/longitudinal_controller.param.yaml
    - source: launch/tier4_control_launch/config/trajectory_follower/lateral_controller.param.yaml
      dest: autoware_launch/config/tier4_control_launch/trajectory_follower/lateral_controller.param.yaml

    # tier4_planning_launch
    - source: launch/tier4_planning_launch/config/scenario_planning/common/common.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/common/common.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/common/nearest_search.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/common/nearest_search.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/common/motion_velocity_smoother/Analytical.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/common/motion_velocity_smoother/Analytical.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/common/motion_velocity_smoother/L2.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/common/motion_velocity_smoother/L2.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/common/motion_velocity_smoother/JerkFiltered.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/common/motion_velocity_smoother/JerkFiltered.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/common/motion_velocity_smoother/Linf.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/common/motion_velocity_smoother/Linf.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/common/motion_velocity_smoother/motion_velocity_smoother.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/common/motion_velocity_smoother/motion_velocity_smoother.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/parking/freespace_planner/freespace_planner.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/parking/freespace_planner/freespace_planner.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/occlusion_spot.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/occlusion_spot.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/intersection.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/intersection.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/traffic_light.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/traffic_light.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/run_out.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/run_out.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/detection_area.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/detection_area.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/virtual_traffic_light.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/virtual_traffic_light.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/crosswalk.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/crosswalk.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/behavior_velocity_planner.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/behavior_velocity_planner.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/stop_line.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/stop_line.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/no_stopping_area.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/no_stopping_area.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/blind_spot.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/blind_spot.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/behavior_path_planner.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/behavior_path_planner.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/pull_over/pull_over.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/pull_over/pull_over.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/pull_out/pull_out.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/pull_out/pull_out.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/lane_following/lane_following.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/lane_following/lane_following.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/side_shift/side_shift.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/side_shift/side_shift.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/avoidance/avoidance.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/avoidance/avoidance.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/lane_change/lane_change.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/lane_change/lane_change.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/rtc_auto_approver/rtc_auto_approver.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/rtc_auto_approver/rtc_auto_approver.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/lane_change_planner/lane_change_planner.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/behavior_planning/lane_change_planner/lane_change_planner.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/motion_planning/obstacle_cruise_planner/obstacle_cruise_planner.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/motion_planning/obstacle_cruise_planner/obstacle_cruise_planner.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/obstacle_stop_planner.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/obstacle_stop_planner.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/adaptive_cruise_control.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/adaptive_cruise_control.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/motion_planning/obstacle_avoidance_planner/obstacle_avoidance_planner.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/motion_planning/obstacle_avoidance_planner/obstacle_avoidance_planner.param.yaml
    - source: launch/tier4_planning_launch/config/scenario_planning/lane_driving/motion_planning/surround_obstacle_checker/surround_obstacle_checker.param.yaml
      dest: autoware_launch/config/tier4_planning_launch/scenario_planning/lane_driving/motion_planning/surround_obstacle_checker/surround_obstacle_checker.param.yaml

    # tier4_perception_launch
    - source: launch/tier4_perception_launch/config/object_recognition/detection/object_lanelet_filter.param.yaml
      dest: autoware_launch/config/tier4_perception_launch/object_recognition/detection/object_lanelet_filter.param.yaml
    - source: launch/tier4_perception_launch/config/object_recognition/detection/pointcloud_map_filter.param.yaml
      dest: autoware_launch/config/tier4_perception_launch/object_recognition/detection/pointcloud_map_filter.param.yaml
    - source: launch/tier4_perception_launch/config/object_recognition/detection/object_position_filter.param.yaml
      dest: autoware_launch/config/tier4_perception_launch/object_recognition/detection/object_position_filter.param.yaml
    - source: launch/tier4_perception_launch/config/object_recognition/tracking/multi_object_tracker/data_association_matrix.param.yaml
      dest: autoware_launch/config/tier4_perception_launch/object_recognition/tracking/multi_object_tracker/data_association_matrix.param.yaml
    - source: launch/tier4_perception_launch/config/obstacle_segmentation/ground_segmentation/ground_segmentation.param.yaml
      dest: autoware_launch/config/tier4_perception_launch/obstacle_segmentation/ground_segmentation/ground_segmentation.param.yaml

    # tier4_system_launch
    - source: launch/tier4_system_launch/config/emergency_handler/emergency_handler.param.yaml
      dest: autoware_launch/config/tier4_system_launch/emergency_handler/emergency_handler.param.yaml
    - source: launch/tier4_system_launch/config/system_monitor/ntp_monitor.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_monitor/ntp_monitor.param.yaml
    - source: launch/tier4_system_launch/config/system_monitor/net_monitor.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_monitor/net_monitor.param.yaml
    - source: launch/tier4_system_launch/config/system_monitor/mem_monitor.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_monitor/mem_monitor.param.yaml
    - source: launch/tier4_system_launch/config/system_monitor/gpu_monitor.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_monitor/gpu_monitor.param.yaml
    - source: launch/tier4_system_launch/config/system_monitor/hdd_monitor.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_monitor/hdd_monitor.param.yaml
    - source: launch/tier4_system_launch/config/system_monitor/process_monitor.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_monitor/process_monitor.param.yaml
    - source: launch/tier4_system_launch/config/system_monitor/cpu_monitor.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_monitor/cpu_monitor.param.yaml
    - source: launch/tier4_system_launch/config/diagnostic_aggregator/vehicle.param.yaml
      dest: autoware_launch/config/tier4_system_launch/diagnostic_aggregator/vehicle.param.yaml
    - source: launch/tier4_system_launch/config/ad_service_state_monitor/ad_service_state_monitor.planning_simulation.param.yaml
      dest: autoware_launch/config/tier4_system_launch/ad_service_state_monitor/ad_service_state_monitor.planning_simulation.param.yaml
    - source: launch/tier4_system_launch/config/ad_service_state_monitor/ad_service_state_monitor.param.yaml
      dest: autoware_launch/config/tier4_system_launch/ad_service_state_monitor/ad_service_state_monitor.param.yaml
    - source: launch/tier4_system_launch/config/system_error_monitor/system_error_monitor.planning_simulation.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_error_monitor/system_error_monitor.planning_simulation.param.yaml
    - source: launch/tier4_system_launch/config/system_error_monitor/system_error_monitor.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_error_monitor/system_error_monitor.param.yaml
    - source: launch/tier4_system_launch/config/system_error_monitor/diagnostic_aggregator/_empty.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_error_monitor/diagnostic_aggregator/_empty.param.yaml
    - source: launch/tier4_system_launch/config/system_error_monitor/diagnostic_aggregator/map.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_error_monitor/diagnostic_aggregator/map.param.yaml
    - source: launch/tier4_system_launch/config/system_error_monitor/diagnostic_aggregator/sensing.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_error_monitor/diagnostic_aggregator/sensing.param.yaml
    - source: launch/tier4_system_launch/config/system_error_monitor/diagnostic_aggregator/control.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_error_monitor/diagnostic_aggregator/control.param.yaml
    - source: launch/tier4_system_launch/config/system_error_monitor/diagnostic_aggregator/localization.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_error_monitor/diagnostic_aggregator/localization.param.yaml
    - source: launch/tier4_system_launch/config/system_error_monitor/diagnostic_aggregator/vehicle.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_error_monitor/diagnostic_aggregator/vehicle.param.yaml
    - source: launch/tier4_system_launch/config/system_error_monitor/diagnostic_aggregator/planning.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_error_monitor/diagnostic_aggregator/planning.param.yaml
    - source: launch/tier4_system_launch/config/system_error_monitor/diagnostic_aggregator/perception.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_error_monitor/diagnostic_aggregator/perception.param.yaml
    - source: launch/tier4_system_launch/config/system_error_monitor/diagnostic_aggregator/system.param.yaml
      dest: autoware_launch/config/tier4_system_launch/system_error_monitor/diagnostic_aggregator/system.param.yaml

    # tier4_simulator_launch
    - source: launch/tier4_simulator_launch/config/fault_injection.param.yaml
      dest: autoware_launch/config/tier4_simulator_launch/fault_injection.param.yaml

    # tier4_integration_launch
```

Reply:
Thanks! I changed the code to the proposed one (with a minor fix).